sklearn/utils/estimator_checks.py

```diff
@@ -1339,6 +1344,31 @@ def check_estimators_dtypes(name, estimator_orig):
                 getattr(estimator, method)(X_train)
 
 
+def check_estimators_preserve_dtypes(name, estimator_orig):
+
```

Comment:
Please skip this test for all estimators that currently fail,
```py
if name in [...]:
   raise SkipTest('Known failure to preserve dtypes')
```
so we can merge this.

The XFAIL support from https://github.com/scikit-learn/scikit-learn/pull/16306 could be used once that PR is merged.

Reply:
They are already skipped here for those who fail. https://github.com/scikit-learn/scikit-learn/pull/16290/files#diff-a95fe0e40350c536a5e303e87ac979c4R194-R213